### DISCUSSION TITLE
Setting Mask Mode to Inpaint not masked seems to affect mask detection quality

### DISCUSSION THREAD

**Discussion by vbmark** [light-and-ray/sd-webui-replacer#92] [Q&A] [answered]
With Mask Mode set to Inpaint not masked, and in the Detection prompt field enter 'face', the quality of the detection seems to diminish.

I'm not entering this as a bug because I'm not sure if maybe there are other settings I should change when using not masked.

Is anyone else using not masked and having to tweak other settings to get it to mask properly?

EDIT: I think the issue is with Detection / Mask Expand.  It seems to be flipped for not masked such that I need to change the value to a negative number.

Thank you.

**Comment by light-and-ray** [comment, marked as answer]
You don't need to set it negative, it changes its sign automatically. I think the problem in prompt, face - is not something that should be selected by segment anything perfectly, because it has very abstract contours

Maybe setting mask negative still can help you, but things outside the face wi) he selected. But better is to draw mask manually in custom mask tab

**Reply by vbmark** [reply]
OK, I understand.  Thank you.
